grim_test.go

```diff
@@ -212,6 +212,35 @@ func TestHookGetsLogged(t *testing.T) {
 
 }
 
+func TestShouldSkip(t *testing.T) {
```

Comment:
This would technically work but its horribly convoluted.
```
func shouldSkip(hook *hookEvent) *string {
	var message *string
	switch {
	case hook.Deleted:
		message = getStringPtr("because it was on a deleted branch")
	case hook.EventName == "push":
	case hook.EventName != "pull_request":
		message = getStringPtr("because the event was not push or pull_request")
	case hook.Action == "opened":
	case hook.Action == "reopened":
	case hook.Action == "synchronize":
	default:
		message = getStringPtr(fmt.Sprintf("because the action: %q was not 'opened', 'reopened', or 'synchronize'", hook.Action))
	}
	return message
}
```